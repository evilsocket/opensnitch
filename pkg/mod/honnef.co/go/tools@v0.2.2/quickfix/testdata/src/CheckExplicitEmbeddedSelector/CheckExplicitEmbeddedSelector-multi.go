package pkg

type MultiLevel struct{ BasicOuter }

func fnMulti() {
	var multi MultiLevel
	_ = multi.BasicOuter.BasicInner.F1 // want `could remove embedded field "BasicOuter" from selector` `could remove embedded field "BasicInner" from selector` `could simplify selectors`
	_ = multi.BasicOuter.F1            // want `could remove embedded field "BasicOuter" from selector`
	_ = multi.BasicInner.F1            // want `could remove embedded field "BasicInner" from selector`
	_ = multi.F1                       // minimal form
}
