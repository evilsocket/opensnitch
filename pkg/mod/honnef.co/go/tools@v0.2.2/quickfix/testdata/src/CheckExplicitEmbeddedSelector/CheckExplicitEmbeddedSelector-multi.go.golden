-- Remove embedded field "BasicInner" from selector --
package pkg

type MultiLevel struct{ BasicOuter }

func fnMulti() {
	var multi MultiLevel
	_ = multi.BasicOuter.F1 // want `could remove embedded field "BasicOuter" from selector` `could remove embedded field "BasicInner" from selector` `could simplify selectors`
	_ = multi.BasicOuter.F1 // want `could remove embedded field "BasicOuter" from selector`
	_ = multi.F1            // want `could remove embedded field "BasicInner" from selector`
	_ = multi.F1            // minimal form
}

-- Remove embedded field "BasicOuter" from selector --
package pkg

type MultiLevel struct{ BasicOuter }

func fnMulti() {
	var multi MultiLevel
	_ = multi.BasicInner.F1 // want `could remove embedded field "BasicOuter" from selector` `could remove embedded field "BasicInner" from selector` `could simplify selectors`
	_ = multi.F1            // want `could remove embedded field "BasicOuter" from selector`
	_ = multi.BasicInner.F1 // want `could remove embedded field "BasicInner" from selector`
	_ = multi.F1            // minimal form
}

-- Remove all embedded fields from selector --
package pkg

type MultiLevel struct{ BasicOuter }

func fnMulti() {
	var multi MultiLevel
	_ = multi.F1            // want `could remove embedded field "BasicOuter" from selector` `could remove embedded field "BasicInner" from selector` `could simplify selectors`
	_ = multi.BasicOuter.F1 // want `could remove embedded field "BasicOuter" from selector`
	_ = multi.BasicInner.F1 // want `could remove embedded field "BasicInner" from selector`
	_ = multi.F1            // minimal form
}
